### ISSUE TITLE
[BUG] Mouse crashes into sides of screen on sway

### ISSUE THREAD

**Issue by sheepymeh** [hmlendea/gfn-electron#174]
### Describe the bug

I am running gfn-electron in sway. The cursor of the system continues to move even when it is captured by gfn-electron. When I move my view far enough, the cursor crashes into the sides of the screen and I am unable to continue moving my view any further.

### To Reproduce

1. Open a game
2. Pan to the left/right until you are unable to do so any further
3. Press Esc and notice that the system cursor is on the edge of the screen

### Operating System

Arch Linux

### Desktop Environment

sway 1.7

### Display Server

Wayland

### Installation method

AUR

### Version

1.10.0

### Is this a fresh install of the app or an update from a past version?

Fresh install

### Did this issue appear right away upon installation/updating, or spontaneously?

Yes, right after installation

### Additional context

This issue is not present when playing directly in the Chromium browser. I am using the ``--enable-features=UseOzonePlatform --ozone-platform=wayland`` flags to run gfn-electron and Chromium in Wayland directly.

**Comment by pil-in-a** [comment]
I have similar problem on my system (fedora 37, KDE, Wayland) and latest flatpak version (update 23/05/03). Mouse crashes to edges and games are overall sluggish and unplayable. It may be related to fact, that I can no longer activate fullscreen with F11 (in Library - it solves some issues with escape key and mouse input), which i was used to in versions before.

**Comment by rby-blackruby** [comment]
@pil-in-a I have the same issue on fedora 38 gnome wayland. What do you mean by downgrading flatpak?

**Comment by pil-in-a** [comment]
@rby-blackruby ... I downgraded `io.github.hmlendea.geforcenow-electron` flatpak package to older version (2023-03-25 instead of 2023-05-03). [Here](https://itsfoss.com/downgrade-flatpak-packages/) is how I did it.

**Comment by hmlendea** [comment]
Is this still an issue for you in the latest version?

**Comment by pil-in-a** [comment]
Fullscreen not working properly is still an issue in the lastest release (cannot manualy activate it via F11, mouse crashing to edge/going to second monitor). Performance seems to be lot better though - playable (still not as smooth as last working version).

**Comment by Victor239** [comment]
Yeah I've got the same issue since a month or so, have been using Chromium instead as well.

**Comment by benjamin051000** [comment]
Same here. Fedora 38 GNOME Wayland running latest gfn as of writing.

**Comment by P0cket-Sevens** [comment]
Similar problem with Ubuntu 22.04 on Gnome42.  Mouse seems to go over the edge and I can't rotate the characters anymore.  Downgraded to the 5/3/23 package from flatpak and that version works just fine.

**Comment by Madame-Louise** [comment]
Same problem here, on Ubuntu 23.10, Wayland, app version 2.0.0 .

**Comment by jobukkit** [comment]
Also have this problem on KaOS with Wayland.

**Comment by benjamin051000** [comment]
I was able to fix this bug, like others, by downgrading the package. This command should work for everyone: `sudo flatpak update --commit=195a02e06f57c1fe201d5321bc75530e78eb153f6e04dc4092978a1fd2e84b5a io.github.hmlendea.geforcenow-electron`

And you should see this: 
```
flatpak info io.github.hmlendea.geforcenow-electron

GeForce NOW Electron - Unofficial GeForce NOW application written in Electron

          ID: io.github.hmlendea.geforcenow-electron
         Ref: app/io.github.hmlendea.geforcenow-electron/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 1.10.0
     License: GPL-3.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 227.9 MB
     Runtime: org.freedesktop.Platform/x86_64/22.08
         Sdk: org.freedesktop.Sdk/x86_64/22.08

      Commit: 195a02e06f57c1fe201d5321bc75530e78eb153f6e04dc4092978a1fd2e84b5a
      Parent: f6680dc1d42925d3fabab0088d8250bf4a809271912984ab2a2d8c3faf7c17e4
     Subject: Update runtime to 22.08 (4b30a22d)
        Date: 2023-03-25 14:26:33 +0000
```

**Comment by Llama037** [comment]
This issue still isn't fixed, is it going to be fixed?

**Comment by MrRossDoughty** [comment]
> I was able to fix this bug, like others, by downgrading the package. This command should work for everyone: `sudo flatpak update --commit=195a02e06f57c1fe201d5321bc75530e78eb153f6e04dc4092978a1fd2e84b5a io.github.hmlendea.geforcenow-electron`
> 
> And you should see this:
> 
> ```
> flatpak info io.github.hmlendea.geforcenow-electron
> 
> GeForce NOW Electron - Unofficial GeForce NOW application written in Electron
> 
>           ID: io.github.hmlendea.geforcenow-electron
>          Ref: app/io.github.hmlendea.geforcenow-electron/x86_64/stable
>         Arch: x86_64
>       Branch: stable
>      Version: 1.10.0
>      License: GPL-3.0
>       Origin: flathub
>   Collection: org.flathub.Stable
> Installation: system
>    Installed: 227.9 MB
>      Runtime: org.freedesktop.Platform/x86_64/22.08
>          Sdk: org.freedesktop.Sdk/x86_64/22.08
> 
>       Commit: 195a02e06f57c1fe201d5321bc75530e78eb153f6e04dc4092978a1fd2e84b5a
>       Parent: f6680dc1d42925d3fabab0088d8250bf4a809271912984ab2a2d8c3faf7c17e4
>      Subject: Update runtime to 22.08 (4b30a22d)
>         Date: 2023-03-25 14:26:33 +0000
> ```
> 
> The fix works on both Fedora 38 and 39 for me.

Recently installed Debian 12 and this is no longer a possible fix as the version quoted is no longer found.

Seems like everything before the version 2.0.0 update has been removed!

**Comment by MrRossDoughty** [comment]
**Workaround**

I have been able to test and if you download the version **1.10.0** Appimage from here:

[https://github.com/hmlendea/gfn-electron/releases/download/v1.1.10/geforcenow-electron_1.1.10_linux.AppImage](
https://github.com/hmlendea/gfn-electron/releases/download/v1.1.10/geforcenow-electron_1.1.10_linux.AppImage
)

Then it works fine.

**Comment by MrHeroj21** [comment]
This issue is still not fixed, having the same problem with the latest version on my fresh KDE Neon with Wayland.

**Comment by swade2569** [comment]
Fedora 40 here with latest Electron version, fully reproducible as noted.

**Comment by MrRossDoughty** [comment]
Thanks for that, I'll give it a go with AppImage.

**Comment by ManuelEscala1998** [comment]
I just found a simple solution for this. The AppImage is good, but I don't like to sign in every single time because I need to do the 2FA this is the other alternative to install the 1.10 because the commit is no longer available. You just need to download the tar.gz file and install it, then you're good to go https://aur.archlinux.org/cgit/aur.git/commit/?h=geforcenow-electron&id=37c3cb1f787253529ae93eba52fd717fe8b9ada0

**Comment by hmlendea** [comment]
Please re-open if this issue still affects the latest version of this app.
